common/scripts/kind_provisioner.sh

```diff
@@ -183,13 +183,22 @@ EOF
   fi
 
   # Create KinD cluster
-  if ! (kind create cluster --name="${NAME}" --config "${CONFIG}" -v4 --retain --image "${IMAGE}" --wait=180s); then
+  if ! (kind create cluster --name="${NAME}" --config "${CONFIG}" -v4 --retain --image "${IMAGE}" ); then
     echo "Could not setup KinD environment. Something wrong with KinD setup. Exporting logs."
     return 9
   fi
   # Workaround kind issue causing taints to not be removed in 1.24
   kubectl taint nodes "${NAME}"-control-plane node-role.kubernetes.io/control-plane- || true
 
+  # If using the ambient config we'll need to install our own CNI
+  # TODO: Will need updating when mc ambient tests are created if we want our own CNI
+  if [[ "$CONFIG" = *"ambient-sc.yaml" ]]; then
+    echo "Setting up ambient cluster, Calico CNI will be used."
+    CONFIG_DIR="$(dirname "$CONFIG")" 
+    kubectl create -f "${CONFIG_DIR}"/calico-install.yaml
+    kubectl create -f "${CONFIG_DIR}"/calico-config.yaml
```

Comment:
Can we actually fetch the versioned Helm artifact and install it? I like that a little better than a giant YAML that's hard to source back to a Calico-published artifact.

Reply:
I can look into using helm instead of doing a `kubectl yeet` on 25k lines of yaml.

Reply:
in general I would like to vendor. for others we do this like metallb as plain yaml, calico is more bloated though

Reply:
yeah, its a **_lot_** of yaml. FWIW that ball of yaml is  https://raw.githubusercontent.com/projectcalico/calico/v3.27.0/manifests/tigera-operator.yaml

The config file isn't a copy of anything it required a bit of work to get this whole train rolling.

Reply:
https://raw.githubusercontent.com/projectcalico/calico/v3.27.0/manifests/calico.yaml is the non-operator variant, its "only" 5k  lines

Reply:
Thanks @howardjohn, This may be a better fit for our needs.

Reply:
This does work better. I've pushed a new version which uses it. I also added some doc about where the file comes from and how it was altered to fit our needs as well as which images we'll mirror when we need to update.